src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -18,7 +18,7 @@ private ReadFilterLibrary(){ /*no instance*/ }
     public static final ReadFilter PASSES_VENDOR_QUALITY_CHECK = read -> ! read.failsVendorQualityCheck();
     public static final ReadFilter MAPPING_QUALITY_AVAILABLE = read -> read.getMappingQuality() != QualityUtils.MAPPING_QUALITY_UNAVAILABLE;
     public static final ReadFilter MAPPING_QUALITY_NOT_ZERO = read -> read.getMappingQuality() != 0;
-    public static final ReadFilter READLENGTH_EQUALS_CIGARLENGTH = read -> read.getLength() == read.getCigar().getReadLength();
+    public static final ReadFilter READLENGTH_EQUALS_CIGARLENGTH = read -> read.isUnmapped() || read.getLength() == read.getCigar().getReadLength();
```

Comment:
Why add this `isUnmapped()` call?  `read.getLength()` is the number of bases in the read, not the alignment length on the reference.  As far as I can see there isn't a reason unmapped reads should be exempted from this check.